### PR TITLE
Redesign for practicality

### PR DESCRIPTION
**Pull request by quantum9Innovation**
Add convenient text box and copy/paste mechanism, fix IP address related errors.
See more details in the commits below.

### CODE REVIEW

**Review by Quantalabs — APPROVED** [review]
Fix merge conflicts

**Comment by quantum9Innovation** [comment]
@Quantalabs ship to shape :anchor: :package:

**Comment by quantum9Innovation** [comment]
@Quantalabs officially good to merge

**Comment by quantum9Innovation** [comment]
@Quantalabs still waiting on merge status

File: index.js [hunk 1]
```diff
@@ -1,174 +1,272 @@
...
+// Imports (external)
+const formidable = require("formidable");
+const mime = require("mime-types");
 const path = require("path");
-const qr = require("qrcode");
-var ip = require("ip");
 const open = require("open");
+const arg = require("arg");
+const qr = require("qrcode");
+const mv = require("mv");
```
Comment (Quantalabs) [inline index.js]: This isn't accurate.

```javascript
// Imports (native)
const http = require("http")
const fs = require("fs");
const path = require("path");

// Imports (external)
const formidable = require("formidable");
const mime = require("mime-types");
const open = require("open");
const arg = require("arg");
const qr = require("qrcode");
const mv = require("mv");
```

**Review by Quantalabs — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (quantum9Innovation) [reply index.js]: Resolved

File: package.json [hunk 2]
```diff
@@ -1,32 +1,44 @@
...
   "scripts": {
-    "test": "mocha"
+    "test": "mocha",
+    "lint": "npx eslint --fix .",
+    "format": "npx prettier --write ."
   },
   "repository": {
     "type": "git",
-    "url": "git+https://github.com/Quantalabs/pastebin.git"
+    "url": "git+https://github.com/quantum9innovation/pastebin.git"
```
Comment (Quantalabs) [inline package.json]:
```suggestion
    "url": "git+https://github.com/Quantalabs/pastebin.git"
```
No, this is my project.

File: package.json [hunk 3]
```diff
@@ -1,32 +1,44 @@
...
   "keywords": [
     "copy",
-    "paste"
+    "paste",
+    "pastebin",
+    "file-transfer",
+    "sync"
   ],
-  "author": "Quantalabs (@Qlabs)",
+  "author": "quantum9innovation (q9i)",
```
Comment (Quantalabs) [inline package.json]: Nope.
```suggestion
  "author": "Quantalabs (@QLabs)",
```

File: package.json [hunk 4]
```diff
@@ -1,32 +1,44 @@
...
   "license": "GPL-3.0-or-later",
   "bugs": {
-    "url": "https://github.com/Quantalabs/pastebin/issues"
+    "url": "https://github.com/quantum9innovation/pastebin/issues"
```
Comment (Quantalabs) [inline package.json]:
```suggestion
    "url": "https://github.com/Quantalabs/pastebin/issues"
```

File: package.json [hunk 5]
```diff
@@ -1,32 +1,44 @@
...
   },
-  "homepage": "https://github.com/Quantalabs/pastebin#readme",
+  "homepage": "https://github.com/quantum9innovation/pastebin#readme",
```
Comment (Quantalabs) [inline package.json]:
```suggestion
  "homepage": "https://github.com/Quantalabs/pastebin#readme",
```

File: target/npmlist.json [hunk 6]
```diff
@@ -0,0 +1 @@
+{"version":"1.0.0b","name":"pastebin","dependencies":{"formidable":{"version":"2.0.1"},"ip":{"version":"1.1.5"},"qrcode":{"version":"1.5.0"}}}
```
Comment (Quantalabs) [inline target/npmlist.json:1]: Delete this whole file- what is this?

File: templates/home.html [hunk 7]
```diff
@@ -1,43 +1,137 @@
...
+      text-align: center;
+    }
+
+    .file {
+      position: relative;
+      display: inline-block;
+      margin: 0 auto;
+      border: 0px;
+      padding: 0px;
+      text-align: center;
```
Comment (Quantalabs) [inline templates/home.html]:
```suggestion
```
You already added this to `body`

File: templates/home.html [hunk 8]
```diff
@@ -1,43 +1,137 @@
...
+      max-width: 800px;
+    }
+
+    #go {
+      float: left;
+      width: 10%;
+      height: 50px;
+      padding: 0px;
+      line-height: 50px;
+      text-align: center;
```
Comment (Quantalabs) [inline templates/home.html]:
```suggestion
```

File: templates/home.html [hunk 9]
```diff
@@ -1,43 +1,137 @@
...
+    }
+
+    input {
+      margin-left: auto;
+      margin-right: auto;
+      text-align: center;
```
Comment (Quantalabs) [inline templates/home.html]:
```suggestion
```

File: templates/home.html [hunk 10]
```diff
@@ -1,43 +1,137 @@
...
+
+    #label {
+      position: absolute;
+      top: 0;
+      left: 0;
+      border-radius: 5px;
+      width: 100%;
+      height: 50px;
+      line-height: 50px;
+      text-align: center;
```
Comment (Quantalabs) [inline templates/home.html]:
```suggestion
```

File: templates/home.html [hunk 11]
```diff
@@ -1,43 +1,137 @@
...
+    .paste {
+      width: 100%;
+      margin: auto;
+    }
+
+    #bin {
+      float: left;
+      margin-left: 25%;
+      margin-right: 10px;
+      text-align: center;
```
Comment (Quantalabs) [inline templates/home.html]:
```suggestion
```

File: templates/home.html [hunk 12]
```diff
@@ -1,43 +1,137 @@
...
+      width: 40%;
+      height: 50px;
+      padding: 5px;
+      line-height: 40px;
+      max-width: 800px;
+      resize: none;
+      white-space: nowrap;
+      overflow-y: hidden;
```
Comment (Quantalabs) [inline templates/home.html]:
```suggestion
      overflow-y: auto;
```

File: templates/home.html [hunk 13]
```diff
@@ -1,43 +1,137 @@
...
+
+    #sep:before {
+      margin-right: 10px;
+      margin-left: 25%;
+    }
+
+    #sep:after {
+      margin-left: 10px;
+      margin-right: 25%;
+    }
```
Comment (Quantalabs) [inline templates/home.html:105]: Move this css to a new file, and change the pastebin script to load the css with the html. This is too much for one file.

File: templates/pasted.html [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+    <style>
+        body {
+          text-align: center;
+        }
+
+        #bin {
+          float: left;
+          margin-left: 25%;
+          margin-right: 10px;
+          text-align: center;
```
Comment (Quantalabs) [inline templates/pasted.html]:
```suggestion
```

File: templates/pasted.html [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+          position: relative;
+          float: left;
+          width: 5vw;
+          min-width: 0.5in;
+          margin-top: -22.5px;
+          height: 50px;
+          padding: 0px;
+          padding-left: 5px;
+          line-height: 50px;
+          text-align: center;
```
Comment (Quantalabs) [inline templates/pasted.html]:
```suggestion
```

File: templates/pasted.html [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+          margin-right: 10px;
+          text-align: center;
+          width: 40%;
+          height: 50px;
+          padding: 5px;
+          line-height: 40px;
+          max-width: 800px;
+          resize: none;
+          white-space: nowrap;
+          overflow-y: hidden;
```
Comment (Quantalabs) [inline templates/pasted.html]:
```suggestion
          overflow-y: auto;
```

File: templates/pasted.html [hunk 17]
```diff
@@ -0,0 +1,92 @@
...
+          padding: 0px;
+          padding-left: 5px;
+          line-height: 50px;
+          text-align: center;
+        }
+
+        input {
+          margin-left: auto;
+          margin-right: auto;
+          text-align: center;
```
Comment (Quantalabs) [inline templates/pasted.html]:
```suggestion
```

File: templates/pasted.html [hunk 18]
```diff
@@ -0,0 +1,92 @@
...
+
+    let doCopy = () => {
+      
+      let text = document.getElementById("bin").value
+      let textarea = document.createElement("textarea")
+      textarea.value = text
+      document.body.appendChild(textarea)
+      textarea.select()
+      document.execCommand("copy")
+      textarea.remove()
```
Comment (Quantalabs) [inline templates/pasted.html]:
```suggestion
      navigator.clipboard.writeText(textarea.value).then(function() {return;}, function(err) {
          console.error('Could not copy text: ', err);
      });
```

`document.execCommand` is [deprecated](https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand)

File: templates/pasted.html [hunk 19]
```diff
@@ -0,0 +1,92 @@
...
+        }
+
+        .icon {
+          position: absolute;
+          display: inline-block;
+          margin: auto;
+          height: 25px;
+          top: 12.5px;
+          left: calc(2.5vw - (25px / 2));
+        }
```
Comment (Quantalabs) [inline templates/pasted.html:55]: Also make this a new file

**Review by Quantalabs — CHANGES_REQUESTED** [review]
Unnecessary CSS and files, and CSS should be moved to new file. Couple errors with the JS, and faulty package.json info.

File: target/npmlist.json (hunk 6 above)
Comment (quantum9Innovation) [reply target/npmlist.json:1]: Looks cool so we're keeping it

File: templates/home.html (hunk 13 above)
Comment (quantum9Innovation) [reply templates/home.html:105]: Too many files = more load time

File: templates/pasted.html (hunk 19 above)
Comment (quantum9Innovation) [reply templates/pasted.html:55]: enough with the files man

**Review by Quantalabs — APPROVED** [review]
lgtm :shipit:

**Pull request merged**